prediction_market_agent_tooling/deploy/betting_strategy.py

```
@@ -176,8 +218,68 @@ def calculate_trades(
+
+        tokens_to_buy = get_buy_outcome_token_amount(
+            bet_amount, buy_direction, yes, no, fee
+        )
+
+        actual_price = bet_amount / tokens_to_buy
+        # price_impact should always be > 0
+        price_impact = (actual_price - expected_price) / expected_price
+        return price_impact
+
+    def calculate_price_impact_deviation_from_target_price_impact(b: xDai) -> float:
+        price_impact = self.calculate_price_impact_for_bet_amount(
```

Ruff

Undefined name `self`

(F821)